tools/telemetry-replay/extract-capture-fixture.py

```
@@ -0,0 +1,288 @@
+def read_frames(capture, session_num, t_from, t_to, hz, variables, shards):
+    """Streams matching frames through zstd + jq, downsampled."""
+    keep = ",".join(f'"{v}": .{v}' for v in variables)
+    jq = (
+        f"select(.SessionNum == {session_num}) | "
+        f"select(.SessionTime >= {t_from} and .SessionTime <= {t_to}) | "
+        "{" + keep + "}"
+    )
+    cmd = (
+        f"cd {capture} && zstd -dcq {shards} | jq -c '{jq}' 2>/dev/null"
+    )
+    proc = subprocess.Popen(
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)